packages/components/forms/src/text-input/TextInput.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState, ChangeEvent } from 'react';
+import { BaseInput } from '../base-input';
+import { Box, Flex } from '@contentful/f36-core';
+import { HelpText } from '@contentful/f36-helptext';
+import { CopyButton } from '@contentful/f36-copybutton';
+import getStyles from './TextInput.styles';
+import { TextInputProps } from './types';
+import { cx } from 'emotion';
+
+export const _TextInput = (
+  {
+    className,
+    testId = 'cf-ui-text-input',
+    countCharacters = false,
+    label,
+    id,
+    value,
+    onChange,
+    maxLength,
+    isInvalid,
+    isDisabled,
+    withCopyButton,
+    onCopy,
+    ...otherProps
+  }: TextInputProps,
+  ref: React.Ref<HTMLInputElement>,
+) => {
+  const [valueState, setValueState] = useState<string | undefined>(value);
```

Comment:
```suggestion
  const [valueState, setValueState] = useState(value);
```

TS will infer the type of value here... I think

Reply:
Btw, do we really need a state here?

couldn't we use ref to get the value in the input? I'm not sure either, just wondering